code/modules/watchtower/watchtower.dm

```diff
@@ -0,0 +1,283 @@
+				return FALSE
+
+			if(!do_after(user,30, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+				return
+
+			if(plasteel.use(25))
+				to_chat(user, SPAN_NOTICE("You construct the watchtower railing."))
+				stage = 5
+				update_icon()
+			else
+				to_chat(user, SPAN_NOTICE("You failed to construct the watchtower railing, you need more plasteel sheets in your offhand."))
+
+			return
+		if(5)
+			if (!HAS_TRAIT(W, TRAIT_TOOL_WRENCH))
+				return
+
+			var/obj/item/stack/rods/rods = user.get_inactive_hand()
+			if(!istype(rods))
+				to_chat(user, SPAN_BOLDWARNING("You need metal rods in your offhand to continue construction of the watchtower."))
+				return FALSE
+
+			if(!do_after(user,30, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+				return
+
+			if(rods.use(60))
+				to_chat(user, SPAN_NOTICE("You construct the watchtower support rods."))
+				stage = 6
+				update_icon()
+			else
+				to_chat(user, SPAN_NOTICE("You failed to construct the watchtower support rods, you need more metal rods in your offhand."))
+
+			return
+		if(6)
+			if (!iswelder(W))
+				return
+
+			if(!HAS_TRAIT(W, TRAIT_TOOL_BLOWTORCH))
+				to_chat(user, SPAN_WARNING("You need a stronger blowtorch!"))
+				return
+
+			var/obj/item/stack/sheet/plasteel/plasteel = user.get_inactive_hand()
+			if(!istype(plasteel))
+				to_chat(user, SPAN_BOLDWARNING("You need plasteel sheets in your offhand to continue construction of the watchtower."))
+				return FALSE
+
+			if(!do_after(user,30, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+				return
+
+			if(plasteel.use(25))
+				to_chat(user, SPAN_NOTICE("You complete the watchtower."))
+				stage = 7
+				update_icon()
+
+
+			else
+				to_chat(user, SPAN_NOTICE("You failed to complete the watchtower, you need more plasteel sheets in your offhand."))
+
+			return
+
+/obj/structure/watchtower/attack_hand(mob/user)
+	if(get_turf(user) == locate(x, y-1, z))
+		if(!do_after(user,30, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+			return
+
+		var/turf/actual_turf = locate(x, y+1, z)
+		ADD_TRAIT(user, TRAIT_ON_WATCHTOWER, "watchtower")
+		user.forceMove(actual_turf)
+		user.client.change_view(user.client.view + 2)
+		var/atom/movable/screen/plane_master/roof/roof_plane = user.hud_used.plane_masters["[ROOF_PLANE]"]
+		roof_plane?.invisibility = INVISIBILITY_MAXIMUM
+		for(var/obj/item/weapon/gun/gun in user)
+			gun.add_bullet_traits(list(BULLET_TRAIT_ENTRY_ID("watchtower_arc", /datum/element/bullet_trait_direct_only)))
+	else if(get_turf(user) == locate(x, y+1, z))
+		if(!do_after(user,30, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, BUSY_ICON_BUILD))
+			return
+
+		REMOVE_TRAIT(user, TRAIT_ON_WATCHTOWER, "watchtower")
+		var/turf/actual_turf = locate(x, y-1, z)
+		user.forceMove(actual_turf)
+		user.client.change_view(user.client.view - 2)
+		var/atom/movable/screen/plane_master/roof/roof_plane = user.hud_used.plane_masters["[ROOF_PLANE]"]
+		roof_plane?.invisibility = 0
+		for(var/obj/item/weapon/gun/gun in user)
```

Comment:
if someone has a gun in their bag, or throws their gun off the watchtower, this trait wont be removed I think

Reply:
Updated the bullet trait to automatically delete itself if the user doesn't have the watchtower trait, also made it so any items picked up will get the bullet traits as well